### DISCUSSION TITLE
Conversational Retrieval Agent

### DISCUSSION THREAD

**Discussion by SamuelH77** [FlowiseAI/Flowise#3292] [Q&A]
Hi everyone,

I am new to Flowise but have been eagerly anticipating using it ever since I saw a demo a few months ago. 

I am working on a simple customer service chatbot that is pre-loaded with the company's website information and other documents. Right now, I am thinking of connecting two document stores for the upsert: 1) Cheerio web scraper, and 2) Pinecone index. I will load the client's documents into Pinecone <-- open to other ideas if that is not right! 

My main question is what agent node do you use for these types of chatbots? The "Conversational Retrieval Agent" was the recommended node but I don't it as an option for agent flows or chat flows. What agent node would work best for this type of application if that one is no longer available? 

Thanks!

**Comment by Crispin-Courtenay** [comment]
If your docs are static, you may want to consider a data store such as an S3 Bucket and have pinecone (or vector DB of choice) draw from there. It can make document management less painful. Neo4J is doing some really interesting work in this market, and at a PoC level is inexpensive. 

Out of curiosity, can your bot create issue tickets?

**Reply by SamuelH77** [reply]
Thank you for the tip. I will look into Neo4J. No it does not create issue tickets...perhaps a future development. 

What do you use for the agent node in these RAG systems?

**Reply by Crispin-Courtenay** [reply]
They have done a ton of work here, with a great many resources.  https://neo4j.com/generativeai/

I am using Pinecone at the moment, as all my current projects are prototypes or POC. Generally, GPT-4o, 1o is a possibility soon, but pay attention to the pricing. With a handful of users, it's probably a hundred bucks a month, but at scale, it could make your eyes water.
